openfold/README.md

```diff
@@ -0,0 +1,183 @@
+# OpenFold PyTorch benchmark implementation
+
+This repository defines the reference implementation for the MLPerf HPC OpenFold benchmark.
+
+## Installation
+
+There are 2 options:
+
+1. Build a docker image based on provided Dockerfile (recommended).
+2. Build a local virtual environment using provided scripts.
+
+### Option 1. Build a docker image.
+
+```bash
+docker build -t openfold_pyt -f Dockerfile .
+```
+
+### Option 2. Build a local virtual environment.
+
+```bash
+# build:
+bash scripts/build_local_openfold_venv.sh /path/to/openfold-venv
+
+# activate:
+source scripts/activate_local_openfold_venv.sh /path/to/openfold-venv
+
+# deactivate:
+source scripts/deactivate_local_openfold_venv.sh
+```
+
+## Dataset
+
+### Download original dataset
+
+**You should ignore this and [download already processed dataset and checkpoint for parameter initialization](#download-processed-dataset-and-checkpoint)!**
+
+Note: downloading scripts require `aria2c`, `rsync`, and `AWS CLI`.
+
+To download PDB and OpenProteinSet use the following commands:
+
+```bash
+bash scripts/download_pdb_mmcif.sh /data/pdb_mmcif/original
+bash scripts/download_open_protein_set.sh /data/open_protein_set/original
+```
+
+After downloading completes, check if you have the following directory tree:
+
+```bash
+data
+├── open_protein_set
+│   └── original  # 606 GiB (524k files, 394k dirs)
+│       ├── LICENSE
+│       ├── duplicate_pdb_chains.txt
+│       └── pdb/
+└── pdb_mmcif
+    └── original  # +55 GiB (+200k files)
+        ├── clusters-by-entity-40.txt
+        ├── obsolete.dat
+        └── raw/
+```
+
+### Preprocess original dataset
+
+**You should ignore this and [download already processed dataset and checkpoint for parameter initialization](#download-processed-dataset-and-checkpoint)!**
+
+```bash
+python scripts/preprocess_pdb_mmcif.py \
+--pdb_mmcif_dirpath /data/pdb_mmcif/original \
+--output_dirpath /data/pdb_mmcif/processed \
+--pdb_clusters_by_entity_filepath /data/pdb_mmcif/original/clusters-by-entity-40.txt \
+--pdb_obsolete_filepath /data/pdb_mmcif/original/obsolete.dat
+
+python scripts/preprocess_open_protein_set.py \
+--open_protein_set_dirpath /data/open_protein_set/original/ \
+--output_dirpath /data/open_protein_set/processed \
+--num_shards 10
+```
+
+The above will produce 616 GiB (~1k files), in the following directory tree:
+
+```bash
+data
+├── open_protein_set
+│   └── processed  # 601 GiB (11 files, 1 dir)
+│       └── pdb_alignments/
+└── pdb_mmcif
+    └── processed  # 15 GiB (+1k files)
+        ├── chains.csv
+        ├── obsolete.dat
+        └── dicts/
+```
+
+### Download processed dataset and checkpoint
+
+TODO: add links for download
```

Comment:
Links added.

Multiple checkpoints can create confusion as to which one to use.

I suggest removing `mlperf_hpc_openfold_resumable_checkpoint_v1.pt` and renaming v2 to one of:

- `mlperf_hpc_openfold_resumable_checkpoint.pt` - no version specified in filename

- `mlperf_hpc_openfold_resumable_checkpoint_b518be46.pt` - version number based on first 8 digits of `sha256sum`. This is common PyTorch convention used in PyTorch Hub (e.g. https://download.pytorch.org/models/resnet18-5c106cde.pth)

In the meantime I replaced md5sum with sha256sum in README.md and train.py script because md5 is not considered secure enough nowadays.